# HG changeset patch
# User Rik <rik@octave.org>
# Date 1375737530 25200
#      Mon Aug 05 14:18:50 2013 -0700
# Node ID 6992c1bb4773aac83d6930f5ac03e3f08556de12
# Parent  397bbb215632f642156f31bbc365164d7231a790
svds.m: Initialize flag variable so interpreter doesn't use flag() colormap instead.

* scripts/sparse/svds.m: Initialize flag variable so interpreter doesn't use
flag() colormap instead.

diff --git a/scripts/sparse/svds.m b/scripts/sparse/svds.m
--- a/scripts/sparse/svds.m
+++ b/scripts/sparse/svds.m
@@ -136,16 +136,19 @@ function [u, s, v, flag] = svds (A, k, s
     error ("svds: SIGMA must be a positive real value or the string 'L'");
   endif
 
   [m, n] = size (A);
   max_a = max (abs (nonzeros (A)));
   if (isempty (max_a))
     max_a = 0;
   endif
+  ## Must initialize variable value, otherwise it may appear to interpreter
+  ## that code is trying to call flag() colormap function.
+  flag = 0;  
 
   if (max_a == 0)
     s = zeros (k, 1);  # special case of zero matrix
   else
     if (nargin < 2)
       k = min ([6, m, n]);
     else
       k = min ([k, m, n]);
